flink-runtime/src/main/java/org/apache/flink/runtime/operators/chaining/ChainedAllReduceDriver.java

```diff
@@ -89,7 +89,7 @@ public void collect(IT record) {
 		numRecordsIn.inc();
 		try {
 			if (base == null) {
-				base = objectReuseEnabled ? record : serializer.copy(record);
+				base = serializer.copy(record);
 			} else {
 				base = objectReuseEnabled ? reducer.reduce(base, record) : serializer.copy(reducer.reduce(base, record));
```

Comment:
What if we return the `record` as the result of `reducer.reduce` (with `objectReuseEnabled == true`)? That would also be problematic, right?